syntax = "proto3";

package federated_learning;

service FederatedLearningService {
  rpc RegisterClient() returns ClientInfo {}
  rpc ScheduleLearningIteration(ScheduleRequest) returns (ScheduleResponse) {}
}

message ClientInfo {
  int32 client_id = 1;
  int32 token = 2;
}

// For the first iteration the ScheduleResponse has a store_id = token
message ScheduleResponse {
  repeated string store_ids = 1;
  string party_id = 2;
}

message ScheduleRequest{
  string program_id = 1;
  string user_id = 2;
  int32 batch_size = 3;
  int32 num_parties = 4;
}